src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -733,16 +734,17 @@ public static bool TryCompleteSendTo(SafeCloseSocket socket, ReadOnlySpan<byte>
 
                 if (sent == -1)
                 {
-                    if (errno != Interop.Error.EAGAIN && errno != Interop.Error.EWOULDBLOCK)
+                    if (!successfulSend && errno != Interop.Error.EAGAIN && errno != Interop.Error.EWOULDBLOCK)
                     {
                         errorCode = GetSocketErrorForErrorCode(errno);
                         return true;
                     }
 
-                    errorCode = SocketError.Success;
+                    errorCode = successfulSend ? SocketError.Success : SocketError.WouldBlock;
```

Comment:
Just noticed that this could probably be restructured slightly:
```C#
if (successfulSend)
{
    errorCode = SocketError.Success;
    return false;
}
else if (errno == Interop.Error.EAGAIN)
{
    errorCode = SocketError.WouldBlock;
    return false;
}
else
{
    errorCode = GetSocketErrorForErrorCode(errno);
    return true;
}
```
Note that EAGAIN has the same value as EWOULDBLOCK, so I don't think we need to do the comparison twice.